File: Recursion/5__string_o_permute.scala


// https://www.hackerrank.com/challenges/string-o-permute

object Solution {

    def main(args: Array[String]) {
        io.Source.stdin.getLines.drop(1).toList
            .map(_.grouped(2).map(_.reverse).mkString(""))
            .map(println)
    }
}

File: Recursion/5__string_o_permute.hs

-- https://www.hackerrank.com/challenges/string-mingling/
main :: IO()
main =  fmap (tail. lines) getContents >>= mapM_ (putStrLn. permute)

permute :: String -> String
permute [] = []
permute (x:y:xs) = y:x: permute xs